gst/mqtt/mqttsink.c

```diff
@@ -816,10 +819,8 @@ gst_mqtt_sink_event (GstBaseSink * basesink, GstEvent * event)
 
   switch (type) {
     case GST_EVENT_EOS:
-      g_mutex_lock (&self->mqtt_sink_mutex);
-      self->mqtt_sink_state = SINK_RENDER_EOS;
+      g_atomic_int_set (&self->mqtt_sink_state, SINK_RENDER_EOS);
```

Comment:
Doesn't applying atomic function for mqtt_sink_state require to have all other mqtt_sink_state to use atomic function? (e.g., Line 555). And how about Line 562-564? (these three lines might be not atomic enough)

Reply:
@myungjoo You are absolutely right. All the state-changing statements should be replaced with ```g_atomic_int_set``` (or ``g_atomic_int_get``` in the case of reading). For 562-565, as you mentioned, only one atomical copying of the state to the local variable is enough, I guess. I will revise that mutex-atomic mess. Thanks.